src/taipy/core/data/abstract_file.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime, timedelta
+from os.path import isfile
+from typing import Dict, List, Optional, Set
+
+import modin.pandas as modin_pd
+import pandas as pd
+from sqlalchemy import create_engine, text
+
+from taipy.config.common.scope import Scope
+
+from .._version._version_manager_factory import _VersionManagerFactory
+from ..common.alias import DataNodeId, Edit
+from ..exceptions.exceptions import InvalidExposedType, MissingRequiredProperty, UnknownDatabaseEngine
+from .data_node import DataNode
+
+
+class _AbstractFileDataNode(DataNode):
+    """Abstract base class for data node implementations (CSVDataNode, ParquetDataNode, ExcelDataNode and
+    PickleDataNode) that are file based."""
+
+    __STORAGE_TYPE = "NOT_IMPLEMENTED"
+
+    __EXPOSED_TYPE_PROPERTY = "exposed_type"
+    __EXPOSED_TYPE_NUMPY = "numpy"
+    __EXPOSED_TYPE_PANDAS = "pandas"
+    __EXPOSED_TYPE_MODIN = "modin"
+    __VALID_STRING_EXPOSED_TYPES = [__EXPOSED_TYPE_PANDAS, __EXPOSED_TYPE_NUMPY, __EXPOSED_TYPE_MODIN]
+
+    __PATH_KEY = "path"
+    __DEFAULT_PATH_KEY = "default_path"
+    __HAS_HEADER_PROPERTY = "has_header"
+
+    _REQUIRED_PROPERTIES: List[str] = []
+
+    def __init__(
+        self,
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        name: Optional[str] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        edits: List[Edit] = None,
+        version: str = None,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        properties: Dict = None,
+    ):
+        if properties is None:
+            properties = {}
+
+        if missing := set(self._REQUIRED_PROPERTIES) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
+        # default_value = properties.pop(self.__DEFAULT_DATA_KEY, None)
+        self._path = properties.get(self.__PATH_KEY, properties.get(self.__DEFAULT_PATH_KEY))
+        if self._path is None:
+            self._path = self.__build_path()
+
+        properties[self.__PATH_KEY] = self._path
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            name,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            edits,
+            version or _VersionManagerFactory._build_manager()._get_latest_version(),
+            validity_period,
+            edit_in_progress,
+            **properties,
+        )
+        if not self._last_edit_date and isfile(self._path):
+            self.last_edit_date = datetime.now()  # type: ignore
+
+    def __build_path(self):
+        from taipy.config.config import Config
+
+        dir_path = pathlib.Path(Config.global_config.storage_folder) / "pickles"
```

Comment:
We need to change the "pickle" folder name.

Reply:
True. I've changed the function to properly point to the correct folder.